src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1408,7 +1408,16 @@ void Session::on_integration_failure(const IntegrationException& error, Download
         m_progress.download = m_download_progress;
     }
     logger.error("Failed to integrate downloaded changesets: %1", error.what());
-    m_conn.close_due_to_protocol_error(error.code());
+
+    m_client_error = util::make_optional<IntegrationException>(error);
```

Comment:
should we `REALM_ASSERT(!m_client_error && !m_error_to_send)` here to make sure we aren't overwriting a previous error?